### PR TITLE
move test_abs to device-generic tests

### PR DESCRIPTION
**Pull request by ngimel**
Per title. test_abs used to be marked as slow_test and run on cpu only. Conceptually similar tests are done in TestTorchMathOps, so it's a matter of adding `abs` test there. 2 remaining checks (correct abs for large-valued long tensors, and correct abs for signed zeros) are factored into separate tests.

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -15293,6 +15249,24 @@ def test_complex_unsupported(self, device, dtype):
...
+        bignumber = 2 ** 31 + 1
+        res = torch.tensor([bignumber], device=device, dtype=dtype)
+        self.assertGreater(res.abs()[0], 0)
+
+    @dtypes(torch.float, torch.double)
+    def test_abs_signed_zero(self, device, dtype):
+        # Both abs(0.0) and abs(-0.0) should result in 0.0
+        size = 128 + 1  # pick a large enough number with remainder so that
+        # both vectorized and nonvectorized op is tested
+        inp = torch.zeros(size, device=device, dtype=dtype)
```
Comment (mruberry) [inline test/test_torch.py:15263]: Nit nit: torch.fill fill_value=-0.?

File: test/test_torch.py [hunk 2]
```diff
@@ -16677,7 +16653,8 @@ def __init__(self,
                   ref_backend='scipy', rtol=0.0008, atol=1e-5),
                   _TorchMathTestMeta('digamma',
                   input_fn=_generate_gamma_input, inputargs=[True], ref_backend='scipy',
-                  replace_inf_with_nan=True)]
+                  replace_inf_with_nan=True),
+                  _TorchMathTestMeta('abs', input_fn=_medium_2d, dtypes=_types_no_half, rtol=0., atol=0.)]
```
Comment (mruberry) [inline test/test_torch.py:16657]: CUDA supports abs for half, right? Is that tested?

**Comment by mruberry** [comment]
What's up with XLA? Maybe retest?

File: test/test_torch.py (hunk 2 above)
Comment (ngimel) [reply test/test_torch.py:16657]: The accuracy is tested in TestTensorDeviceOps, testing behavior wrt non-contiguous inputs for half specifically is not particularly useful. test_abs used to run on cpu only.

File: test/test_torch.py (hunk 1 above)
Comment (ngimel) [reply test/test_torch.py:15263]: I'm filling with + and - zeros to make sure they both behave as expected.

**Review by mruberry — APPROVED** [review]
Cool!
